renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -4720,6 +4720,109 @@ void WrappedVulkan::vkCmdEndQueryIndexedEXT(VkCommandBuffer commandBuffer, VkQue
   }
 }
 
+void WrappedVulkan::vkCmdBeginConditionalRenderingEXT(
+    VkCommandBuffer commandBuffer,
+    const VkConditionalRenderingBeginInfoEXT *pConditionalRenderingBegin)
+{
+  SCOPED_DBG_SINK();
+
+  VkConditionalRenderingBeginInfoEXT unwrappedConditionalRenderingBegin = *pConditionalRenderingBegin;
+  unwrappedConditionalRenderingBegin.buffer = Unwrap(unwrappedConditionalRenderingBegin.buffer);
+
+  SERIALISE_TIME_CALL(ObjDisp(commandBuffer)
+                          ->CmdBeginConditionalRenderingEXT(Unwrap(commandBuffer),
+                                                            &unwrappedConditionalRenderingBegin));
+
+  if(IsCaptureMode(m_State))
+  {
+    VkResourceRecord *record = GetRecord(commandBuffer);
+
+    CACHE_THREAD_SERIALISER();
+
+    SCOPED_SERIALISE_CHUNK(VulkanChunk::vkCmdBeginConditionalRenderingEXT);
+    Serialise_vkCmdBeginConditionalRenderingEXT(ser, commandBuffer, pConditionalRenderingBegin);
+
+    record->AddChunk(scope.Get());
+
+    VkResourceRecord *buf = GetRecord(pConditionalRenderingBegin->buffer);
+
+    record->MarkResourceFrameReferenced(buf->GetResourceID(), eFrameRef_Read);
+    record->MarkResourceFrameReferenced(buf->baseResource, eFrameRef_Read);
+
+    if(IsActiveCapturing(m_State))
+      GetResourceManager()->MarkResourceFrameReferenced(buf->GetResourceID(), eFrameRef_Read);
```

Comment:
This call to `MarkResourceFrameReferenced` shouldn't be needed, the ones above in the command buffer's record should be sufficient. Also marking referenced directly in the resource manager side-steps tracking of which command buffers actually get submitted, so it's to be avoided in the general case.

Reply:
You right, it is unnecessary.